subsys/bluetooth/mesh/scheduler_internal.h

```diff
@@ -40,10 +47,17 @@ static inline void scheduler_action_unpack(struct net_buf_simple *buf,
 	entry->scene_number = net_buf_simple_pull_le16(buf);
 }
 
-static inline void scheduler_action_pack(struct net_buf_simple *buf,
+static inline bool scheduler_action_pack(struct net_buf_simple *buf,
 				uint8_t idx,
 				const struct bt_mesh_schedule_entry *entry)
 {
+	if (entry->year > MAX_YEAR || entry->day > MAX_DAY ||
```

Comment:
I do not think this is good idea. If client is able to manage checking by self, it should do it.
You will send some message that will be rejected. It will eat energy to be sent and relayed, and eat energy to get back negative reply. Also, it will generate unnecessary traffic and will cause rpl update and etc etc. 
We need to refuse this if it is possible.